src/enforester.js

```diff
@@ -1313,26 +1337,23 @@ export class Enforester {
 
   enforestNewExpression() {
     this.matchKeyword('new');
-    let callee;
-    if (this.isKeyword(this.peek(), 'new')) {
-      callee = this.enforestNewExpression();
-    } else if (this.isKeyword(this.peek(), 'super')) {
-      callee = this.enforestExpressionLoop();
-    } else if (this.isPunctuator(this.peek(), '.') && this.isIdentifier(this.peek(1), 'target')) {
+    let lookahead = this.peek();
+    if (this.isPunctuator(lookahead, '.') && this.isIdentifier(this.peek(1), 'target')) {
       this.advance();
       this.advance();
       return new Term('NewTargetExpression', {});
     } else {
-      callee = new Term('IdentifierExpression', { name : this.enforestIdentifier() });
+      this.term = this.enforestLeftHandSideExpression({ allowCall: false });
```

Comment:
Yeah. I think that's an artifact of my earlier efforts. I'll fix it ASAP.

On Jul 1, 2016 11:08 AM, "Tim Disney" <notifications@github.com> wrote:

> In src/enforester.js
> <https://github.com/sweet-js/sweet.js/pull/557#discussion_r69322224>:
>
> >        this.advance();
> >        this.advance();
> >        return new Term('NewTargetExpression', {});
> >      } else {
> > -      callee = new Term('IdentifierExpression', { name : this.enforestIdentifier() });
> > +      this.term = this.enforestLeftHandSideExpression({ allowCall: false });
>
> I don't think you need to assign to this.term here, the local callee
> should work better.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/sweet-js/sweet.js/pull/557/files/a3243d486e09b66ae5b332865ee5b63d4294c68a#r69322224>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ACmO9bDO-Ql53-OTJxV_2YzmwQnmqvkPks5qRTtygaJpZM4I7YO0>
> .
>